File: src/models/sql/Event.js
module.exports = (sequelize, DataTypes) => {
  const Event = sequelize.define('Event', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    nama_event: { type: DataTypes.STRING(150), allowNull: false },
    deskripsi: { type: DataTypes.TEXT },
    lokasi: { type: DataTypes.STRING(200) },
    tanggal_mulai: { type: DataTypes.DATEONLY, allowNull: false },
    tanggal_selesai: { type: DataTypes.DATEONLY, allowNull: false },
    status: { type: DataTypes.ENUM('draft', 'aktif', 'selesai', 'batal'), defaultValue: 'draft' },
    ketua_id: { type: DataTypes.INTEGER, allowNull: false },
  }, {
    tableName: 'events',
    timestamps: true,
    createdAt: 'created_at',
    updatedAt: 'updated_at',
  });

  Event.associate = (models) => {
    Event.belongsTo(models.User, { foreignKey: 'ketua_id', as: 'ketua' });
    Event.hasMany(models.Vendor, { foreignKey: 'event_id', as: 'vendors' });
    Event.hasMany(models.Rundown, { foreignKey: 'event_id', as: 'rundowns' });
    Event.hasMany(models.Tugas, { foreignKey: 'event_id', as: 'tugas' });
    Event.hasMany(models.LaporanKetua, { foreignKey: 'event_id', as: 'laporan' });
  };

  return Event;
};

File: mobile/lib/features/events/domain/entities/event.dart
/// Event entity matching backend `events` table.
class Event {
  final int id;
  final String namaEvent;
  final String? deskripsi;
  final String? lokasi;
  final DateTime tanggalMulai;
  final DateTime tanggalSelesai;
  final String status;
  final int ketuaId;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const Event({
    required this.id,
    required this.namaEvent,
    this.deskripsi,
    this.lokasi,
    required this.tanggalMulai,
    required this.tanggalSelesai,
    required this.status,
    required this.ketuaId,
    this.createdAt,
    this.updatedAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] as int,
      namaEvent: json['nama_event'] as String? ?? '',
      deskripsi: json['deskripsi'] as String?,
      lokasi: json['lokasi'] as String?,
      tanggalMulai: DateTime.parse(json['tanggal_mulai'].toString()),
      tanggalSelesai: DateTime.parse(json['tanggal_selesai'].toString()),
      status: json['status'] as String? ?? 'draft',
      ketuaId: json['ketua_id'] as int,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'].toString())
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.tryParse(json['updated_at'].toString())
          : null,
    );
  }
}
